lib/linalg/svd/IncrementalSVDFastUpdate.cpp

```diff
@@ -86,6 +86,18 @@ IncrementalSVDFastUpdate::~IncrementalSVDFastUpdate()
     }
 }
 
+const Matrix*
+IncrementalSVDFastUpdate::getSpatialBasis()
+{
+    // For fast update, we choose to compute the basis
+    // only when we try to get its value. Therefore,
+    // computeBasis() is called inside getSpatialBasis().
+    computeBasis(); // WARNING: this is costly
```

Comment:
`computeBasis` doe not only compute `d_basis`, but also `d_basis_right` and `d_S`. If we defer `computeBasis` until we call basis functions, then we need to ensure `getTemporalBasis` and `getSingularValues` also call `computeBasis`. At the same time, it should be also ensured that `computeBasis` is not called repeatedly, when calling these `get--` functions.

Reply:
`IncrementalSVDFastUpdate` has been restored to the original version, and all the new features have been moved to a new class `IncrementalSVDBrand`. `comupteBasis` now calls `updateSpatialBasis` and `updateTemporalBasis` to compute basis matrices. These `update--` functions are also individually called in each `get--` functions. For example, `getTemporalBasis` calls only `updateTemporalBasis` before returning `d_basis_right` and does not update `d_basis`, which could be costly and unnecessary.